src/umpire/resource/FileMemoryResource.cpp

```diff
@@ -0,0 +1,124 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+
+#include "umpire/resource/FileMemoryResource.hpp"
+
+#include <errno.h>
+#include <string.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <sys/mman.h>
+
+namespace umpire {
+namespace resource {
+
+int FileMemoryResource::s_file_counter {0};
+
+FileMemoryResource::FileMemoryResource(
+    Platform platform, 
+    const std::string& name,
+    int id,
+    MemoryResourceTraits traits) :
+  MemoryResource(name, id, traits),
+  m_platform{platform},
+  m_size_map{}
+{
+} 
+
+FileMemoryResource::~FileMemoryResource()
+{
+  std::vector<void*> leaked_items;
+
+  for ( auto const& m : m_size_map ) {
+      leaked_items.push_back(m.first);
+  }
+
+  for ( auto const& p : leaked_items ) {
+      deallocate(p);
+  }
+}
+
+void* FileMemoryResource::allocate(std::size_t bytes)
+{
+  if (bytes <= 0) { UMPIRE_ERROR( "Bytes Requested Error: Bytes size is less than 1"); }
```

Comment:
Please remove this check since `std::size_t` is an `unsigned` type and can never be < 0.